src/ToolBox/SOS/lldbplugin/debugclient.cpp

```diff
@@ -62,6 +62,18 @@ DebugClient::GetCurrentThread()
     return thread;
 }
 
+lldb::SBFrame 
+DebugClient::GetCurrentFrame()
+{
+    lldb::SBFrame frame;
+    lldb::SBThread thread = GetCurrentThread();
+    if (thread.IsValid())
+    {
+        frame = thread.GetFrameAtIndex(0);
```

Comment:
Yes, it probably should be the current/selected stack frame instead of the top one.

From: Sergiy Kuryata [mailto:notifications@github.com]
Sent: Wednesday, March 18, 2015 10:35 PM
To: dotnet/coreclr
Cc: Mike McLaughlin
Subject: Re: [coreclr] Enable the rest of the sos dump commands and supporting functions. (#500)


In src/ToolBox/SOS/lldbplugin/debugclient.cpp<https://github.com/dotnet/coreclr/pull/500#discussion_r26730632>:

> @@ -62,6 +62,18 @@ DebugClient::GetCurrentThread()

>      return thread;

>  }

>

> +lldb::SBFrame

> +DebugClient::GetCurrentFrame()

> +{

> +    lldb::SBFrame frame;

> +    lldb::SBThread thread = GetCurrentThread();

> +    if (thread.IsValid())

> +    {

> +        frame = thread.GetFrameAtIndex(0);

Is it correct to always use index 0 here?

—
Reply to this email directly or view it on GitHub<https://github.com/dotnet/coreclr/pull/500/files#r26730632>.